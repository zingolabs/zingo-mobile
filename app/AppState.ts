// new class with all the information about the sync process
export class SyncStatusReport {
  // sync ID
  syncID: number;

  // total of batches to process (usually 1000 blocks per batch)
  totalBatches: number;

  // batch that is processing now
  currentBatch: number;

  // total blocks per batch
  blocksPerBatch: number;

  // last block of the wallet
  lastBlockWallet: number;

  // block that is processing
  currentBlock: number;

  inProgress: boolean;

  lastError: string;

  secondsPerBatch: number;

  percent: number;

  message: string;

  process_end_block: number;

  lastBlockServer: number;

  constructor() {
    this.syncID = 0;
    this.totalBatches = 0;
    this.currentBatch = 0;
    this.blocksPerBatch = 1000; // this is set in zingolib
    this.lastBlockWallet = 0;
    this.currentBlock = 0;
    this.inProgress = false;
    this.lastError = '';
    this.secondsPerBatch = 0;
    this.percent = 0;
    this.message = '';
    this.process_end_block = 0;
    this.lastBlockServer = 0;
  }
}

export class TotalBalance {
  // Total t address, confirmed and spendable
  transparentBal: number;

  // Total private, confirmed + unconfirmed
  privateBal: number;

  // Total orchard, confirmed + unconfirmed
  orchardBal: number;

  // Total private, confirmed funds that are spendable
  spendablePrivate: number;

  // Total orchard, confirmed funds that are spendable
  spendableOrchard: number;

  // Total unconfirmed + spendable
  total: number;

  constructor() {
    this.transparentBal = 0;
    this.privateBal = 0;
    this.spendablePrivate = 0;
    this.orchardBal = 0;
    this.spendableOrchard = 0;
    this.total = 0;
  }
}

export class Address {
  uaAddress: string;
  address: string;
  addressKind: string;
  containsPending: boolean;

  constructor(uaAddress: string, address: string, addressKind: string) {
    this.uaAddress = uaAddress;
    this.address = address;
    this.addressKind = addressKind;
    this.containsPending = false;
  }
}

export class AddressBookEntry {
  label: string;

  address: string;

  constructor(label: string, address: string) {
    this.label = label;
    this.address = address;
  }
}

export class SettingsFileEntry {
  server?: string;

  constructor(server: string) {
    this.server = server;
  }
}

export interface TxDetailType {
  address: string;
  amount: number;
  memo: string | null;
}

// List of transactions.
export interface Transaction {
  type: string;
  address: string;
  amount: number;
  position: string;
  confirmations: number;
  txid: string;
  time: number;
  zec_price: number | null;
  detailedTxns: TxDetailType[];
}

// This is the type that the RPC interface expects for sending
export interface SendJsonToType {
  address: string;
  amount: number;
  memo?: string;
}

export class ToAddr {
  id: number;
  to: string;
  amount: string;
  amountUSD: string;
  memo: string;

  constructor(id: number) {
    this.id = id;

    this.to = '';
    this.amount = '';
    this.amountUSD = '';
    this.memo = '';
  }
}

export class SendPageState {
  toaddrs: ToAddr[];

  constructor() {
    this.toaddrs = [];
  }
}

export class ReceivePageState {
  // A newly created address to show by default
  newAddress: string;

  // The key used for the receive page component.
  // Increment to force re-render
  rerenderKey: number;

  constructor() {
    this.newAddress = '';
    this.rerenderKey = 0;
  }
}

export interface InfoType {
  chain_name: string;
  serverUri: string;
  latestBlock: number;
  connections: number;
  version: string;
  verificationProgress: number;
  currencyName: string;
  solps: number;
  zecPrice: number | null;
  defaultFee: number;
}

export class SendProgress {
  sendInProgress: boolean;
  progress: number;
  total: number;
  etaSeconds: number | string;

  constructor() {
    this.sendInProgress = false;
    this.progress = 0;
    this.total = 0;
    this.etaSeconds = 0;
  }
}

export interface ServerSelectState {
  modalIsOpen: boolean;
}

export class PasswordState {
  showPassword: boolean;

  confirmNeeded: boolean;

  passwordCallback: ((password: string) => void) | null;

  closeCallback: (() => void) | null;

  helpText: string | null;

  constructor() {
    this.showPassword = false;
    this.confirmNeeded = false;
    this.passwordCallback = null;
    this.closeCallback = null;
    this.helpText = null;
  }
}

export class ErrorModalData {
  title: string;
  body: string;

  modalIsOpen: boolean;

  constructor() {
    this.modalIsOpen = false;
    this.title = '';
    this.body = '';
  }
}

export interface ConnectedCompanionApp {
  name: string;
  lastSeen: number;
}

export interface WalletSeed {
  seed: string;
  birthday: number;
}

export interface SyncStatus {
  inProgress: boolean;
  progress: number;
  blocks: string;
}

export class WalletSettings {
  download_memos: string;
  transaction_filter_threshold: string;
  server?: string;

  constructor() {
    this.download_memos = 'wallet';
    this.transaction_filter_threshold = '500';
  }
}

export default interface AppState {
  // Info about the current sync process
  syncStatusReport: SyncStatusReport;

  // The total confirmed and unconfirmed balance in this wallet
  totalBalance: TotalBalance;

  // A map type that contains address -> privatekey mapping, for display on the receive page
  // This mapping is ephemeral, and will disappear when the user navigates away.
  addressPrivateKeys: Map<string, string>;

  // List of all addresses in the wallet, including change addresses and addresses
  // that don't have any balance or are unused
  addresses: Address[];

  // List of Address / Label pairs
  addressBook: AddressBookEntry[];

  // List of all T and Z and O transactions
  transactions: Transaction[] | null;

  // The state of the send page, as the user constructs a transaction
  sendPageState: SendPageState;

  // Any state for the receive page
  receivePageState: ReceivePageState;

  // getinfo and getblockchaininfo result
  info: InfoType | null;

  // Is the app rescanning?
  rescanning: boolean;

  // Callbacks for the password dialog box
  //passwordState: PasswordState;

  wallet_settings: WalletSettings;

  syncingStatus: SyncStatus | null;

  // Data for any error or info modal
  errorModalData: ErrorModalData;

  // Build progress from Tx
  txBuildProgress: SendProgress;

  walletSeed: WalletSeed | null;

  isMenuDrawerOpen: boolean;

  selectedMenuDrawerItem: string;

  aboutModalVisible: boolean;

  computingModalVisible: boolean;

  settingsModalVisible: boolean;

  infoModalVisible: boolean;

  rescanModalVisible: boolean;

  seedViewModalVisible: boolean;
  seedChangeModalVisible: boolean;
  seedBackupModalVisible: boolean;
  seedServerModalVisible: boolean;

  syncReportModalVisible: boolean;

  newServer: string | null;

  uaAddress: string | null;

  // eslint-disable-next-line semi
}
