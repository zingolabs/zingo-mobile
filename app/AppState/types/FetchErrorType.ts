export default interface FetchErrorType {
  command: string;
  error: string;

  // eslint-disable-next-line semi
}
