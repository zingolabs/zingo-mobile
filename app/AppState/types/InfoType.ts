export default interface InfoType {
  chain_name: 'main' | 'test' | 'regtest';
  serverUri: string;
  latestBlock: number;
  connections: number;
  version: string;
  verificationProgress: number;
  currencyName: string;
  solps: number;
  defaultFee: number; // here is a float number, like 0.0001
  zingolib: string;
  // eslint-disable-next-line semi
}
