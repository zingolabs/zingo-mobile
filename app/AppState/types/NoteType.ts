export default interface NoteType {
  created_in_block: number;
  datetime: number;
  created_in_txid: string;
  value: number;
  scriptkey?: string;
  unconfirmed?: string;
  is_change: boolean;
  address: string;
  spendable?: string;
  spent: string;
  spent_at_height: number;
  unconfirmed_spent: string;

  pool: 'Orchard' | 'Sapling' | 'Transparent';
  type: 'Pending' | 'Unspent';

  // eslint-disable-next-line semi
}
