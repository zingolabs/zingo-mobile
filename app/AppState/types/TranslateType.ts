export type TranslateType =
  | string
  | string[]
  | { value: string; text: string }[]
  | { value: boolean; text: string }[]
  | { [key: string]: string[] };
