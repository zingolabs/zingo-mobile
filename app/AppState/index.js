import AddressClass from './classes/AddressClass';
import AddressBookClass from './classes/AddressBookClass';
import ErrorModalDataClass from './classes/ErrorModalDataClass';
import ReceivePageStateClass from './classes/ReceivePageStateClass';
import SendPageStateClass from './classes/SendPageStateClass';
import SendProgressClass from './classes/SendProgressClass';
import SettingsFileClass from './classes/SettingsFileClass';
import SyncingStatusClass from './classes/SyncingStatusClass';
import ToAddrClass from './classes/ToAddrClass';
import TotalBalanceClass from './classes/TotalBalanceClass';
import WalletSettingsClass from './classes/WalletSettingsClass';

import InfoType from './types/InfoType';
import SendJsonToTypeType from './types/SendJsonToTypeType';
import TransactionType from './types/TransactionType';
import TxDetailType from './types/TxDetailType';
import WalletType from './types/WalletType';
import { TranslateType } from './types/TranslateType';
import NetInfoType from './types/NetInfoType';
import BackgroundType from './types/BackgroundType';
import BackgroundErrorType from './types/BackgroundErrorType';
import ServerType from './types/ServerType';
import SnackbarType from './types/SnackbarType';
import ZecPriceType from './types/ZecPriceType';
import FetchErrorType from './types/FetchErrorType';

import AppStateLoaded from './AppStateLoaded';
import AppStateLoading from './AppStateLoading';

export {
  AppStateLoaded,
  AppStateLoading,
  AddressClass,
  AddressBookClass,
  ErrorModalDataClass,
  ReceivePageStateClass,
  SendPageStateClass,
  SendProgressClass,
  SettingsFileClass,
  SyncingStatusClass,
  ToAddrClass,
  TotalBalanceClass,
  WalletSettingsClass,
  TransactionType,
  TxDetailType,
  WalletType,
  SendJsonToTypeType,
  InfoType,
  ZecPriceType,
  BackgroundType,
  BackgroundErrorType,
  TranslateType,
  NetInfoType,
  ServerType,
  SnackbarType,
  FetchErrorType,
};
