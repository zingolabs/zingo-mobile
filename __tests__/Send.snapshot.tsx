/**
 * @format
 */

import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';
import Send from '../components/Send';
import { defaultAppStateLoaded, ContextAppLoadedProvider } from '../app/context';
import { ThemeType } from '../app/types';

jest.useFakeTimers();
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: '',
}));
jest.mock('react-native-localize', () => ({
  getNumberFormatSettings: () => {
    return {
      decimalSeparator: '.', // us
      groupingSeparator: ',', // us
    };
  },
}));
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('react-native-reanimated', () => {
  return class Reanimated {
    public static Value() {
      return jest.fn(() => {});
    }
    public static View() {
      return '';
    }
  };
});
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RPCModule = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});
jest.mock('@react-native-community/netinfo', () => {
  const RN = jest.requireActual('react-native');

  RN.NativeModules.RNCNetInfo = {
    execute: jest.fn(() => '{}'),
  };

  return RN;
});
const Theme: ThemeType = {
  dark: true,
  colors: {
    background: '#011401', //'#010101',
    card: '#011401', //'#401717',
    border: '#ffffff',
    primary: '#18bd18', //'#df4100',
    primaryDisabled: '#5a8c5a', //'rgba(90, 140, 90, 1)',
    secondaryDisabled: '#233623',
    text: '#c3c3c3',
    zingo: '#888888',
    placeholder: '#888888',
    money: '#ffffff',
    syncing: '#ebff5a',
    notification: '',
  },
};
jest.mock('@react-navigation/native', () => ({
  useIsFocused: jest.fn(),
  useTheme: () => Theme,
}));
jest.mock('react-native-picker-select', () => 'RNPickerSelect');

// test suite
describe('Component Send - test', () => {
  //snapshot test
  const state = defaultAppStateLoaded;
  state.transactions = [
    {
      type: 'Sent',
      fee: 0.0001,
      confirmations: 22,
      txid: 'sent-txid-1234567890',
      time: Date.now(),
      zec_price: 33.33,
      txDetails: [
        {
          address: 'sent-address-1-12345678901234567890',
          amount: 0.12345678,
          memos: ['hola', '  & ', 'hello'],
        },
        {
          address: 'sent-address-2-09876543210987654321',
          amount: 0,
          memos: ['hello', '  & ', 'hola'],
        },
      ],
    },
    {
      type: 'SendToSelf',
      fee: 0.0001,
      confirmations: 12,
      txid: 'sendtoself-txid-1234567890',
      time: Date.now(),
      zec_price: 33.33,
      txDetails: [
        {
          address: '',
          amount: 0,
          memos: ['orchard memo', 'sapling memo'],
        },
      ],
    },
    {
      type: 'Received',
      confirmations: 133,
      txid: 'receive-txid-1234567890',
      time: Date.now(),
      zec_price: 66.66,
      txDetails: [
        {
          address: '',
          amount: 0.77654321,
          pool: 'Orchard',
          memos: ['hola', '  & ', 'hello'],
        },
        {
          address: '',
          amount: 0.1,
          pool: 'Sapling',
          memos: ['hello', '  & ', 'hola'],
        },
      ],
    },
  ];
  state.uaAddress = 'UA-12345678901234567890';
  state.addresses = [
    {
      uaAddress: 'UA-12345678901234567890',
      address: 'UA-12345678901234567890',
      addressKind: 'u',
      containsPending: false,
      receivers: 'ozt',
    },
    {
      uaAddress: 'UA-12345678901234567890',
      address: 'sapling-12345678901234567890',
      addressKind: 'z',
      containsPending: false,
      receivers: 'z',
    },
    {
      uaAddress: 'UA-12345678901234567890',
      address: 'transparent-12345678901234567890',
      addressKind: 't',
      containsPending: false,
      receivers: 't',
    },
  ];
  state.translate = () => 'text translated';
  state.currency = 'USD';
  state.info.currencyName = 'ZEC';
  state.zecPrice.zecPrice = 33.33;
  state.info.defaultFee = 10000;
  state.totalBalance.total = 1.12345678;
  state.totalBalance.orchardBal = 0.6;
  state.totalBalance.spendableOrchard = 0.3;
  state.totalBalance.privateBal = 0.4;
  state.totalBalance.spendablePrivate = 0.2;
  state.totalBalance.transparentBal = 0.12345678;
  state.sendPageState.toaddr.id = 1234567890;
  state.sendPageState.toaddr.to = 'UA-12345678901234567890';
  state.sendPageState.toaddr.amount = '1.12345678';
  state.sendPageState.toaddr.amountCurrency = '50.22';
  state.sendPageState.toaddr.memo = 'memo';
  const onFunction = jest.fn();

  test('Send no currency, privacy normal & mode basic - snapshot', () => {
    // no currency
    state.currency = '';
    // privacy normal
    state.privacy = false;
    // mode basic
    state.mode = 'basic';
    const send = render(
      <ContextAppLoadedProvider value={state}>
        <Send
          setSendPageState={onFunction}
          sendTransaction={onFunction}
          clearToAddr={onFunction}
          setSendProgress={onFunction}
          toggleMenuDrawer={onFunction}
          setComputingModalVisible={onFunction}
          setZecPrice={onFunction}
          set_privacy_option={onFunction}
          setPoolsToShieldSelectSapling={onFunction}
          setPoolsToShieldSelectTransparent={onFunction}
        />
      </ContextAppLoadedProvider>,
    );
    expect(send.toJSON()).toMatchSnapshot();
  });

  test('Send currency USD, privacy high & mode advanced - snapshot', () => {
    // no currency
    state.currency = 'USD';
    // privacy normal
    state.privacy = true;
    // mode basic
    state.mode = 'advanced';
    const send = render(
      <ContextAppLoadedProvider value={state}>
        <Send
          setSendPageState={onFunction}
          sendTransaction={onFunction}
          clearToAddr={onFunction}
          setSendProgress={onFunction}
          toggleMenuDrawer={onFunction}
          setComputingModalVisible={onFunction}
          setZecPrice={onFunction}
          set_privacy_option={onFunction}
          setPoolsToShieldSelectSapling={onFunction}
          setPoolsToShieldSelectTransparent={onFunction}
        />
      </ContextAppLoadedProvider>,
    );
    expect(send.toJSON()).toMatchSnapshot();
  });
});
