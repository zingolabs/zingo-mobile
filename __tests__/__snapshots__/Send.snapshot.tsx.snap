// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Send - test Send - snapshot 1`] = `
<View
  accessibilityLabel="text translated"
  accessible={true}
  style={
    {
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
      "marginBottom": 200,
      "width": "100%",
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  />
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  />
  <RCTScrollView
    contentContainerStyle={{}}
    keyboardShouldPersistTaps="handled"
    testID="send.scrollView"
  >
    <View>
      <View
        style={
          {
            "marginBottom": 30,
          }
        }
      >
        <View
          style={
            {
              "display": "flex",
              "marginTop": 10,
              "padding": 10,
            }
          }
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#c3c3c3",
                  "fontSize": 18,
                  "fontWeight": "600",
                  "opacity": 1,
                }
              }
            >
              text translated
            </Text>
          </View>
          <View
            style={
              {
                "borderColor": "#c3c3c3",
                "borderRadius": 5,
                "borderWidth": 1,
                "flex": 1,
                "marginTop": 5,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessibilityLabel="text translated"
                accessible={true}
                style={
                  {
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <TextInput
                  editable={true}
                  onChangeText={[Function]}
                  placeholder="text translated"
                  placeholderTextColor="#888888"
                  style={
                    {
                      "color": "#c3c3c3",
                      "fontSize": 16,
                      "fontWeight": "600",
                      "marginLeft": 5,
                    }
                  }
                  testID="send.addressplaceholder"
                  value="UA-12345678901234567890"
                />
              </View>
              <View
                style={
                  {
                    "width": 58,
                  }
                }
              >
                <View
                  accessibilityLabel="text translated"
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                  testID="send.scan-button"
                >
                  <
                    color="#ffffff"
                    icon={
                      {
                        "icon": [
                          448,
                          512,
                          [],
                          "f029",
                          "M48 32C21.5 32 0 53.5 0 80v96c0 26.5 21.5 48 48 48h96c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48H48zm80 64v64H64V96h64zM48 288c-26.5 0-48 21.5-48 48v96c0 26.5 21.5 48 48 48h96c26.5 0 48-21.5 48-48V336c0-26.5-21.5-48-48-48H48zm80 64v64H64V352h64zM256 80v96c0 26.5 21.5 48 48 48h96c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48H304c-26.5 0-48 21.5-48 48zm64 16h64v64H320V96zm32 352v32h32V448H352zm96 0H416v32h32V448zM416 288v32H352V288H256v96 96h64V384h32v32h96V352 320 288H416z",
                        ],
                        "iconName": "qrcode",
                        "prefix": "fas",
                      }
                    }
                    size={48}
                    style={
                      {
                        "margin": 5,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "minHeight": 48,
                  "minWidth": 48,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "borderRadius": 10,
                    "justifyContent": "flex-end",
                    "margin": 0,
                    "minHeight": 48,
                    "minWidth": 48,
                    "padding": 0,
                    "paddingBottom": 3,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#c3c3c3",
                      "opacity": 0.65,
                    }
                  }
                >
                  text translated
                </Text>
              </View>
            </View>
            <Text
              style={
                {
                  "color": "#18bd18",
                }
              }
            >
              text translated
            </Text>
          </View>
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "width": "60%",
                }
              }
            >
              <View
                style={
                  {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#c3c3c3",
                      "fontSize": 20,
                      "fontWeight": "600",
                      "marginRight": 5,
                      "marginTop": 20,
                      "opacity": 1,
                    }
                  }
                >
                  á™‡
                </Text>
                <View
                  accessibilityLabel="text translated"
                  accessible={true}
                  style={
                    {
                      "borderColor": "#c3c3c3",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexGrow": 1,
                      "marginTop": 5,
                      "minHeight": 48,
                      "minWidth": 48,
                      "width": "75%",
                    }
                  }
                >
                  <TextInput
                    editable={true}
                    keyboardType="numeric"
                    maxLength={20}
                    onChangeText={[Function]}
                    onEndEditing={[Function]}
                    placeholder="#.########"
                    placeholderTextColor="#888888"
                    style={
                      {
                        "color": "#c3c3c3",
                        "fontSize": 18,
                        "fontWeight": "600",
                        "marginLeft": 5,
                        "minHeight": 48,
                        "minWidth": 48,
                      }
                    }
                    testID="send.amount"
                    value="1.12345678"
                  />
                </View>
                <Text
                  style={
                    {
                      "color": "#c3c3c3",
                      "fontSize": 18,
                      "fontWeight": "600",
                      "marginLeft": 5,
                      "marginRight": 10,
                      "marginTop": 19,
                      "opacity": 1,
                    }
                  }
                >
                  ZEC
                </Text>
              </View>
              <View
                style={
                  {
                    "display": "flex",
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                      "marginTop": 10,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#c3c3c3",
                        "fontSize": 14,
                        "fontWeight": "600",
                        "opacity": 1,
                      }
                    }
                  >
                    text translated
                  </Text>
                  <View
                    style={
                      {
                        "alignItems": "baseline",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "color": "red",
                          "fontSize": 15,
                        }
                      }
                    >
                      á™‡
                    </Text>
                    <Text
                      style={
                        {
                          "color": "red",
                          "fontSize": 15,
                          "fontWeight": "700",
                        }
                      }
                    >
                       0.0000
                    </Text>
                    <Text
                      style={
                        {
                          "color": "red",
                          "fontSize": 7.5,
                          "paddingBottom": 0,
                        }
                      }
                    >
                      0000
                    </Text>
                    <Text
                      style={
                        {
                          "color": "red",
                          "fontSize": 15,
                        }
                      }
                    >
                       ZEC
                    </Text>
                  </View>
                </View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "backgroundColor": "#011401",
                        "borderRadius": 10,
                        "display": "flex",
                        "flexDirection": "row",
                        "marginTop": 5,
                        "padding": 5,
                      }
                    }
                  >
                    <
                      color="#18bd18"
                      icon={
                        {
                          "icon": [
                            192,
                            512,
                            [],
                            "f129",
                            "M144 80c0 26.5-21.5 48-48 48s-48-21.5-48-48s21.5-48 48-48s48 21.5 48 48zM0 224c0-17.7 14.3-32 32-32H96c17.7 0 32 14.3 32 32V448h32c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H64V256H32c-17.7 0-32-14.3-32-32z",
                          ],
                          "iconName": "info",
                          "prefix": "fas",
                        }
                      }
                      size={14}
                    />
                    <Text
                      style={
                        {
                          "color": "#c3c3c3",
                          "opacity": 0.65,
                        }
                      }
                    >
                      text translated
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "column",
                  "justifyContent": "flex-start",
                  "width": "35%",
                }
              }
            >
              <View
                style={
                  {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#c3c3c3",
                      "fontSize": 18,
                      "fontWeight": "600",
                      "marginRight": 5,
                      "marginTop": 15,
                      "opacity": 1,
                    }
                  }
                >
                  $
                </Text>
                <View
                  accessibilityLabel="text translated"
                  accessible={true}
                  style={
                    {
                      "borderColor": "#c3c3c3",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "flexGrow": 1,
                      "marginTop": 5,
                      "minHeight": 48,
                      "minWidth": 48,
                      "width": "55%",
                    }
                  }
                >
                  <TextInput
                    editable={true}
                    keyboardType="numeric"
                    maxLength={15}
                    onChangeText={[Function]}
                    onEndEditing={[Function]}
                    placeholder="#.##"
                    placeholderTextColor="#888888"
                    style={
                      {
                        "color": "#c3c3c3",
                        "fontSize": 18,
                        "fontWeight": "600",
                        "marginLeft": 5,
                        "minHeight": 48,
                        "minWidth": 48,
                      }
                    }
                    value="50.22"
                  />
                </View>
                <Text
                  style={
                    {
                      "color": "#c3c3c3",
                      "fontSize": 18,
                      "fontWeight": "600",
                      "marginLeft": 5,
                      "marginTop": 15,
                      "opacity": 1,
                    }
                  }
                >
                  USD
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "column",
                  }
                }
              >
                <View
                  style={
                    {
                      "alignItems": "baseline",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#ffffff",
                        "fontSize": 13,
                        "marginTop": 10,
                      }
                    }
                  >
                    $
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#ffffff",
                        "fontSize": 13,
                        "fontWeight": "700",
                        "marginTop": 10,
                      }
                    }
                  >
                     0.00
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#ffffff",
                        "fontSize": 13,
                        "marginTop": 10,
                      }
                    }
                  >
                     USD
                  </Text>
                </View>
                <View
                  style={
                    {
                      "marginLeft": 5,
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#011401",
                          "borderRadius": 10,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "margin": 0,
                          "minHeight": 40,
                          "minWidth": 40,
                          "padding": 5,
                        }
                      }
                    >
                      <
                        color="#18bd18"
                        icon={
                          {
                            "icon": [
                              512,
                              512,
                              [
                                128472,
                                "refresh",
                                "sync",
                              ],
                              "f021",
                              "M105.1 202.6c7.7-21.8 20.2-42.3 37.8-59.8c62.5-62.5 163.8-62.5 226.3 0L386.3 160H336c-17.7 0-32 14.3-32 32s14.3 32 32 32H463.5c0 0 0 0 0 0h.4c17.7 0 32-14.3 32-32V64c0-17.7-14.3-32-32-32s-32 14.3-32 32v51.2L414.4 97.6c-87.5-87.5-229.3-87.5-316.8 0C73.2 122 55.6 150.7 44.8 181.4c-5.9 16.7 2.9 34.9 19.5 40.8s34.9-2.9 40.8-19.5zM39 289.3c-5 1.5-9.8 4.2-13.7 8.2c-4 4-6.7 8.8-8.1 14c-.3 1.2-.6 2.5-.8 3.8c-.3 1.7-.4 3.4-.4 5.1V448c0 17.7 14.3 32 32 32s32-14.3 32-32V396.9l17.6 17.5 0 0c87.5 87.4 229.3 87.4 316.7 0c24.4-24.4 42.1-53.1 52.9-83.7c5.9-16.7-2.9-34.9-19.5-40.8s-34.9 2.9-40.8 19.5c-7.7 21.8-20.2 42.3-37.8 59.8c-62.5 62.5-163.8 62.5-226.3 0l-.1-.1L125.6 352H176c17.7 0 32-14.3 32-32s-14.3-32-32-32H48.4c-1.6 0-3.2 .1-4.8 .3s-3.1 .5-4.6 1z",
                            ],
                            "iconName": "arrows-rotate",
                            "prefix": "fas",
                          }
                        }
                        size={20}
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "flexGrow": 1,
              "justifyContent": "center",
              "marginVertical": 5,
            }
          }
        >
          <View
            accessibilityLabel="title text translated"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "rgba(90, 140, 90, 1)",
                "borderColor": "#18bd18",
                "borderRadius": 10,
                "borderWidth": 2,
                "justifyContent": "center",
                "maxWidth": "90%",
                "minHeight": 48,
                "minWidth": "30%",
                "opacity": 1,
                "padding": 0,
                "paddingLeft": 20,
                "paddingRight": 20,
              }
            }
            testID="send.button"
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 0,
                  "minHeight": 48,
                  "minWidth": 48,
                  "padding": 0,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#011401",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "textTransform": "uppercase",
                  }
                }
              >
                text translated
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "borderColor": "#18bd18",
                "borderRadius": 10,
                "borderWidth": 2,
                "justifyContent": "center",
                "marginLeft": 10,
                "maxWidth": "90%",
                "minHeight": 48,
                "minWidth": "30%",
                "opacity": 1,
                "padding": 0,
                "paddingLeft": 20,
                "paddingRight": 20,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "margin": 0,
                  "minHeight": 48,
                  "minWidth": 48,
                  "padding": 0,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#18bd18",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "textTransform": "uppercase",
                  }
                }
              >
                text translated
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
